### ISSUE TITLE
wxListBook: Position of AssignImageList()

### ISSUE THREAD

**Issue by HackiWimmer** [wxWidgets/wxWidgets#22364]
I'm using wx 3.1.6 und Windows 10.

If an image list of wxListbook controls (and other book ctrls) is managed in this manner

	wxImageList* imageList = new wxImageList(16, 16, true);
	m_listBook->AssignImageList(imageList);

	imageList->Add(ImageLibPins().Bitmap("BMP_A"));
	imageList->Add(ImageLibPins().Bitmap("BMP_B"));
	imageList->Add(ImageLibPins().Bitmap("BMP_C"));

the bitmaps are not visible!

If the order is changed to the following

	wxImageList* imageList = new wxImageList(16, 16, true);
	imageList->Add(ImageLibPins().Bitmap("BMP_A"));
	imageList->Add(ImageLibPins().Bitmap("BMP_B"));
	imageList->Add(ImageLibPins().Bitmap("BMP_C"));

	m_listBook->AssignImageList(imageList);

the bitmaps are available/visible.

With wx 3.1.5 on Windows 10 this still works in both cases. 
Some code generaters like wxCrafter prefer the first version.

Is this a changed behaviour or a bug?

**Comment by vadz** [comment]
This wasn't supposed to change, thanks for reporting this. I'll try to make the former version work again.

**Comment by HackiWimmer** [comment]
For the meanwhile: As a work around something like that works:


wxImageList* imgListOld = listBook->GetImageList();
	
const wxSize size = imgListOld->GetSize();
wxImageList* imgListNew = new wxImageList(size.GetWidth(), size.GetHeight());
	
const int count = imgListOld->GetImageCount();
for (int i = 0; i<count; i++ )
	imgListNew->Add(imgListOld->GetBitmap(i));
	
listBook->AssignImageList(imgListNew);

**Comment by vadz** [comment]
Does this actually happen with the other book controls? If I make these change to the notebook sample:
```diff
diff --git a/samples/notebook/notebook.cpp b/samples/notebook/notebook.cpp
index e4eb4c9bb3..472a9fc7b3 100644
--- a/samples/notebook/notebook.cpp
+++ b/samples/notebook/notebook.cpp
@@ -554,7 +557,10 @@ void MyFrame::RecreateBook()
     // wxToolbook doesn't work without icons so always use them for it.
     if ( m_chkShowImages || m_type == Type_Toolbook )
     {
-        m_bookCtrl->SetImages(m_images);
+        auto iml = new wxImageList(32, 32);
+        m_bookCtrl->AssignImageList(iml);
+        for ( auto const& im: m_images )
+            iml->Add(im.GetBitmap(wxSize(32, 32)));
     }

     if ( oldBook )
```
I do see the problem with `wxListbook`, but not any other classes. Could you please tell what are the other ones that don't show the images for you and under which platform does it happen?

**Comment by vadz** [comment]
The problem with `wxListbook` is/should be fixed by #22372, please let me know if you see anything wrong with the other controls or still see any problems with this one even after applying the changes from that PR.

TIA!
